src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,129 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(requests.get("findById"))) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Book book = null;
+            if (resultSet.next()) {
+                book = parseResultSet(resultSet);
+            }
+            return Optional.ofNullable(book);
+        } catch (SQLException e) {
+            throw new DataProcessingException(connectionErrorMsg(), e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(requests.get("findAll"))) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> result = new ArrayList<>();
+            while (resultSet.next()) {
+                result.add(parseResultSet(resultSet));
+            }
+            return result;
+        } catch (SQLException e) {
+            throw new DataProcessingException(connectionErrorMsg(), e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(requests.get("update"))) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            if (!isRowAffected(statement.executeUpdate())) {
+                throw new DataProcessingException("Update failed for book with id:" + book.getId());
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException(connectionErrorMsg(), e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(requests.get("deleteById"))) {
+            statement.setLong(1, id);
+            return isRowAffected(statement.executeUpdate());
+        } catch (SQLException e) {
+            throw new DataProcessingException(connectionErrorMsg(), e);
+        }
+    }
+
+    private Book parseResultSet(ResultSet resultSet) throws SQLException {
+        Book book = new Book();
+        book.setId((Long) resultSet.getObject("id"));
+        book.setTitle(resultSet.getString("title"));
+        book.setPrice(resultSet.getBigDecimal("price"));
+        return book;
+    }
+
+    private String connectionErrorMsg() {
+        return "Connection to the DB failed";
+    }
+
+    private boolean isRowAffected(int affectedRow) {
+        return affectedRow > 0;
+    }
```

Comment:
this method seems to be redundant as well

```
if(!isRowAffected(statement.executedUpdate())
```
is much more difficult to read than 
```
if(statement.executeUpdate() != 1)
or
if(statement.executeUpdate() < 1)
```
therefore you can remove this method and simplify your `if` statements